src/components/Admin/StudentGroupPanel/index.tsx

```diff
@@ -4,41 +4,126 @@ import styles from './styles.module.scss';
 import { observer } from 'mobx-react-lite';
 import { useStore } from '@tdev-hooks/useStore';
 import Button from '@tdev-components/shared/Button';
-import { mdiPlusCircleOutline } from '@mdi/js';
+import { mdiMagnify, mdiPlusCircleOutline, mdiRestore } from '@mdi/js';
 import StudentGroup from '@tdev-components/StudentGroup';
 import _ from 'es-toolkit/compat';
-import scheduleMicrotask from '@tdev-components/util/scheduleMicrotask';
 import { action } from 'mobx';
+import Icon from '@mdi/react';
 
 const StudentGroupPanel = observer(() => {
     const userStore = useStore('userStore');
     const groupStore = useStore('studentGroupStore');
     const current = userStore.current;
+    const [searchFilter, setSearchFilter] = React.useState('');
+    const [searchRegex, setSearchRegex] = React.useState(new RegExp(searchFilter, 'i'));
+
+    React.useEffect(() => {
+        setSearchRegex(new RegExp(searchFilter, 'i'));
+    }, [searchFilter]);
```

Comment:
either use `useMemo` or ditch the caching since we don't expect more than 20? or 100? student groups?

With memo it would be (optionally added here error handling for invalid regex inputs?)

```suggestion
    const searchRegeg = React.useMemo(() => {
        return new RegExp(searchFilter, 'i');
    }, [searchFilter]}
```